lib/command/common/run-with-cover.js

```diff
@@ -166,6 +172,8 @@ function run(args, commandName, enableHooks, callback) {
                     postLoadHookFile = path.resolve(postRequireHook);
                 } else if (opts.yui) { //EXPERIMENTAL code: do not rely on this in anyway until the docs say it is allowed
                     postLoadHookFile = path.resolve(__dirname, '../../util/yui-load-hook');
+                } else if (opts['source-map']) {
+                    hookOpts.postLoadHook = require('../../util/source-map-support')(instrumenter, matchFn);
```

Comment:
Not a fan of this being handled differently, but there was a need to access the `instrumenter` instance and changed the hook API seemed like it would be even more problematic.